tests/integration/pilot/workloadentry_test.go

```diff
@@ -0,0 +1,141 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+)
+
+func TestWorkloadEntry(t *testing.T) {
+	// nolint: staticcheck
+	framework.NewTest(t).
+		Features("traffic.reachability").
+		Run(func(t framework.TestContext) {
+			// Define a simulated EW gateway
```

Comment:
Do we need a simulated one? the framework always deploys a real one anyways

Reply:
I think the comment verbiage is probably misleading. It's really only simulated in that it's a single-cluster test. I will update to clarify on that.